File: tkpdreputation/src/main/java/com/tokopedia/tkpd/tkpdreputation/data/source/CloudGetLikeDislikeDataSource.java
package com.tokopedia.tkpd.tkpdreputation.data.source;

import com.tokopedia.core.base.domain.RequestParams;
import com.tokopedia.core.network.apiservices.user.ReputationService;
import com.tokopedia.tkpd.tkpdreputation.domain.model.GetLikeDislikeReviewDomain;
import com.tokopedia.tkpd.tkpdreputation.data.mapper.GetLikeDislikeMapper;

import rx.Observable;

/**
 * @author by nisie on 9/29/17.
 */

public class CloudGetLikeDislikeDataSource {
    private final ReputationService reputationService;
    private final GetLikeDislikeMapper getLikeDislikeMapper;

    public CloudGetLikeDislikeDataSource(ReputationService reputationService,
                                         GetLikeDislikeMapper getLikeDislikeMapper) {

        this.reputationService = reputationService;
        this.getLikeDislikeMapper = getLikeDislikeMapper;
    }


    public Observable<GetLikeDislikeReviewDomain> getLikeDislikeReview(RequestParams requestParams) {
        return reputationService.getApi()
                .getLikeDislikeReview(requestParams.getParameters())
                .map(getLikeDislikeMapper);
    }
}

File: features/merchant/review_shop/src/main/java/com/tokopedia/review_shop/shop/data/source/CloudGetLikeDislikeDataSource.kt
package com.tokopedia.review_shop.shop.data.source

import com.tokopedia.review_shop.shop.data.mapper.GetLikeDislikeMapper
import com.tokopedia.review_shop.shop.data.network.ReputationService
import com.tokopedia.review_shop.shop.domain.model.GetLikeDislikeReviewDomain
import com.tokopedia.usecase.RequestParams
import rx.Observable

/**
 * @author by nisie on 9/29/17.
 */
class CloudGetLikeDislikeDataSource(private val reputationService: ReputationService,
                                    private val getLikeDislikeMapper: GetLikeDislikeMapper) {
    fun getLikeDislikeReview(requestParams: RequestParams): Observable<GetLikeDislikeReviewDomain> {
        return reputationService.api!!
                .getLikeDislikeReview(requestParams.parameters)!!
                .map(getLikeDislikeMapper)
    }

}